python/graphstorm/gconstruct/README.md

```diff
@@ -1,34 +1,69 @@
-To load a user-defined graph to GraphStorm, a user needs to format their data as follows.
-Specifially, each node/edge type should be stored in separate folders. The data associated
+To load a user-defined graph to GraphStorm, a user needs to provide a JSON file to describe
+their data. Users need to store the node/edge data of different types in separate files.
+The data associated
 with a node/edge type can be stored in multiple files. Below shows an example of the graph
 data with two node types and two edge types. In this example, the node/edge data are stored
-in the parquet format. We support three input formats: parquet, CSV and JSON.
-```
-data_root_dir/
-  |-- input_data.json
-  |-- node1/
-  |   |-- node_1_1.parquet
-  |   |-- node_1_2.parquet
-  |-- node2/
-  |   |-- node_2_1.parquet
-  |   |-- node_2_2.parquet
-  |-- edge1/
-  |   |-- edge_1_1.parquet
-  |   |-- edge_1_2.parquet
-  |-- edge2/
-  |   |-- edge_2_1.parquet
-  |   |-- edge_2_2.parquet
-```
+in the parquet format.
+
+The JSON file that describes the graph data defines where to get node data
+and edge data to construct a graph. Below shows an example of such a JSON file.
+In the highest level, it contains two fields: `nodes` and `edges`.
+
+`nodes` contains a list of node types and the information of a node type
+is stored in a dictionary. A node dictionary contains multiple fields and
+most fields are optional:
+* `node_type` specifies the node type. This field is mandatory.
+* `files` specifies the input files for the node data. This field is mandatory.
+There are multiple options to specify the input files.
+For a single input file, it contains the path of a single file.
+For multiple files, it contains the path of files with a wildcard,
+or a list of file paths.
+* `format` specifies the input file format. This field is mandatory.
+Currently, the pipeline supports two formats: parquet and JSON.
+The detailed format information is specified in the format section.
+* `node_id_col` specifies the column that contains the node IDs. This field is optional.
```

Comment:
if two dicts provide the node features, one of the dicts should contain the node IDs while the other doesn't have node IDs. otherwise, we need to check if the two node IDs are consistent. There is a check in the code to verify if every node type has an ID mapping generated from the node IDs in the input file. If a node type doesn't have node IDs, it will report an error.